src/Emulator/Peripherals/Peripherals/Timers/OpenTitan_Timer.cs

```diff
@@ -0,0 +1,201 @@
+                                this.Log(LogLevel.Noisy, $"Set timer0 compare lower 0x{value:X} - CompareValue0: 0x{compareValue0:X}");
+                             }, valueProviderCallback: _ =>
+                             {
+                                this.Log(LogLevel.Noisy, $"Get timer0 compare lower - CompareValue0: 0x{compareValue0:X}");
+                                return (uint)(compareValue0 & (ulong)0x00000000_FFFFFFFF);
+                             })},
+
+                 {(long)Registers.CompareUpper0, new DoubleWordRegister(this)
+                 .WithValueField(0, 32, name: "timer compare upper", writeCallback: (_, value) =>
+                         {
+                            compareValue0 &= (ulong)0x00000000_FFFFFFFF;
+                            compareValue0 |= ((ulong)value << 32);
+                            Compare = compareValue0;
+                            this.Log(LogLevel.Noisy, $"Set timer0 compare upper 0x{value:X} - CompareValue0: 0x{compareValue0:X}");
+                         }, valueProviderCallback: _ =>
+                         {
+                            this.Log(LogLevel.Noisy, $"Get timer0 compare upper - CompareValue0: 0x{compareValue0:X}");
+                            return (uint)((compareValue0 >> 32) & (ulong)0x00000000_FFFFFFFF);
+                         })},
+
+                 {(long)Registers.InterruptEnable0, new DoubleWordRegister(this)
+                 .WithFlag(0, out timerInterruptEnable0, name: "timer0 interrupt enable", changeCallback: (_, value) =>
+                         {
+                            string enable_message = value ? "Enabled" : "Disabled";
+                            this.Log(LogLevel.Noisy, $"{enable_message} timer0 interrupt");
+                            EventEnabled = value;
+                         })
+                 .WithIgnoredBits(1, 31)},
+
+                 {(long)Registers.InterruptState0, new DoubleWordRegister(this)
+                 .WithFlag(0, FieldMode.Read | FieldMode.Write, name: "timer0 innerupt state", writeCallback: (_, value) =>
+                         {
+                            string message = value ? "Request Clear" : "Unset";
+                            this.Log(LogLevel.Noisy, $"{message} timer0 interrupt");
+                            if (value)
+                            {
+                                timerExpired0IRQ.Set(false);
+                            }
+                         }, valueProviderCallback: _ =>
+                         {
+                            return timerExpired0IRQ.IsSet;
+                         })
+                 .WithIgnoredBits(1, 31)},
+           };
+           registers = new DoubleWordRegisterCollection(this, registersMap);
+
+        }
+
+        public override void Reset()
+        {
+            base.Reset();
+            timerExpired0IRQ.Set(false);
+        }
+
+        public GPIO IRQ { get; } = new GPIO();
```

Comment:
how is this IRQ related to the `timerExpired0IRQ`? Is this one used at all/needed?